lib/git_worktree.rb

```diff
@@ -10,19 +10,23 @@ def initialize(options = {})
     require 'rugged'
 
     raise ArgumentError, "Must specify path" unless options.key?(:path)
-    @path          = options[:path]
-    @email         = options[:email]
-    @username      = options[:username] || ""
-    @bare          = options[:bare]
-    @commit_sha    = options[:commit_sha]
-    @password      = options[:password] || ""
-    @fast_forward_merge = options[:ff] || true
-    @remote_name   = 'origin'
-    @cred          = Rugged::Credentials::UserPassword.new(:username => @username,
-                                                           :password => @password)
-    @credentials_set = false
-    @base_name = File.basename(@path)
+    @path                 = options[:path]
+    @email                = options[:email]
+    @username             = options[:username]
+    @bare                 = options[:bare]
+    @commit_sha           = options[:commit_sha]
+    @password             = options[:password]
+    @ssh_private_key      = options[:ssh_private_key]
+    @fast_forward_merge   = options[:ff] || true
     @certificate_check_cb = options[:certificate_check]
+
+    @remote_name = 'origin'
+    @base_name   = File.basename(@path)
+
+    if @ssh_private_key && !Rugged.features.include?(:ssh)
```

Comment:
In our appliance builds I've already talked to @simaishi about getting SSH support for rugged and it's relatively simple.  If a local developer doesn't compile rugged with ssh support this gives them a nice message.

FWIW, the command I used to build it locally on a Mac is:

```
brew install libssh2
PKG_CONFIG_PATH="$PKG_CONFIG_PATH:/usr/local/opt/openssl/lib/pkgconfig" gem install rugged -v 0.27.7
```

On Fedora/CentOS, the presence of libssh2 seems to be enough to get it to build properly.

We will have to document this somewhere in the dev setup.  Even so, a "normal" installation of rugged will still work for user/pass over http(s)...just ssh won't work.